test/integration/resourcemanager/managedresource/reconciler_suite_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reconciler_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/resourcemanager/controller/managedresource"
+	"github.com/gardener/gardener/pkg/resourcemanager/predicate"
+	managerpredicate "github.com/gardener/gardener/pkg/resourcemanager/predicate"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap/zapcore"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func TestManagedResourceController(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "ManagedResource Reconciler Integration Test Suite")
+}
+
+const namespaceName = "test-namespace"
+
+var (
+	ctx       = context.Background()
+	mgrCancel context.CancelFunc
+
+	logger     logr.Logger
+	testEnv    *envtest.Environment
+	restConfig *rest.Config
+	testClient client.Client
+
+	namespace *corev1.Namespace
+
+	err error
```

Comment:
I guess this err var could be moved to be a local var in the `BeforeSuite`. It does not makes sense to have it as a pkg var. (I assume that you "inherited" this from existing env test, but let's fix this occurrence)

```suggestion
```